kernel/src/schema/mod.rs

```diff
@@ -331,9 +331,9 @@ impl InvariantChecker {
     ///
     /// This traverses the entire schema to check for the presence of the "delta.invariants"
     /// metadata key.
-    pub(crate) fn has_invariants(schema: &Schema) -> bool {
+    pub(crate) fn has_invariants(schema: SchemaRef) -> bool {
         let mut checker = InvariantChecker::default();
-        let _ = checker.transform_struct(schema);
+        let _ = checker.transform_struct(schema.as_ref());
         checker.has_invariants
```

Comment:
Since we are already borrowing at the callsite, would it make sense to just keep `&Schema` as argument type here?

Reply:
Returned back